### PR TITLE
Migrate rest test suite to https

### PR DESCRIPTION
**Pull request by fviale**
- Start the rest server in https when running tests and adapt test code to use insecure mode
 - DataSpaceClient and SchedulerEventReceiver: reuse the provided security configuration (secure/insecure) from the SchedulerClient
 - LoadPackage.groovy: use insecure mode on scheduler client

### CODE REVIEW

**Comment by fviale** [comment]
Retest this please

File: rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerEventReceiver.java [hunk 1]
```diff
@@ -94,7 +99,15 @@ public void start() throws IOException {
     private void openAndReceive(final SchedulerEventListener eventListener, boolean myEventsOnly,
             SchedulerEvent... events) throws IOException {
         Client client = ClientFactory.getDefault().newClient();
-        socket = client.create(client.newOptionsBuilder().reconnect(false).build());
+        AsyncHttpClientConfig.Builder builder = new AsyncHttpClientConfig.Builder();
+        if (insecure) {
+            builder = builder.setAcceptAnyCertificate(true)
+                             .setHostnameVerifier(SSLConnectionSocketFactory.ALLOW_ALL_HOSTNAME_VERIFIER);
```
Comment (sonarqube-activeeon) [inline rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerEventReceiver.java]: MINOR Remove this use of "ALLOW_ALL_HOSTNAME_VERIFIER"; it is deprecated. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerEventReceiver.java [hunk 2]
```diff
@@ -123,7 +136,11 @@ public void on(Object t) {
 
     public void stop() {
         if (socket != null) {
-            socket.close();
+            try {
+                socket.close();
+            } catch (Exception ignored) {
```
Comment (sonarqube-activeeon) [inline rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerEventReceiver.java]: MAJOR Either remove or fill this block of code. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00108)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 247 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 245 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SchedulerClient.java#L348](https://github.com/ow2-proactive/scheduling/blob/0847d6d22d00e113a2a37bd795bb78f014b70d43/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java#L348): Catch Exception instead of Throwable. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)
1. ![MINOR][MINOR] [ISchedulerClient.java#L106](https://github.com/ow2-proactive/scheduling/blob/0847d6d22d00e113a2a37bd795bb78f014b70d43/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L106): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L125](https://github.com/ow2-proactive/scheduling/blob/0847d6d22d00e113a2a37bd795bb78f014b70d43/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L125): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L146](https://github.com/ow2-proactive/scheduling/blob/0847d6d22d00e113a2a37bd795bb78f014b70d43/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L146): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L166](https://github.com/ow2-proactive/scheduling/blob/0847d6d22d00e113a2a37bd795bb78f014b70d43/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L166): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L186](https://github.com/ow2-proactive/scheduling/blob/0847d6d22d00e113a2a37bd795bb78f014b70d43/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L186): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L209](https://github.com/ow2-proactive/scheduling/blob/0847d6d22d00e113a2a37bd795bb78f014b70d43/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L209): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L230](https://github.com/ow2-proactive/scheduling/blob/0847d6d22d00e113a2a37bd795bb78f014b70d43/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L230): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L247](https://github.com/ow2-proactive/scheduling/blob/0847d6d22d00e113a2a37bd795bb78f014b70d43/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L247): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L258](https://github.com/ow2-proactive/scheduling/blob/0847d6d22d00e113a2a37bd795bb78f014b70d43/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L258): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by fviale** [comment]
Retest this please

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 247 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 245 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [SchedulerClient.java#L348](https://github.com/ow2-proactive/scheduling/blob/3a7a2b2a71a94c2b474b2a32f2ce1cb44b0833bb/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/SchedulerClient.java#L348): Catch Exception instead of Throwable. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1181)
1. ![MINOR][MINOR] [ISchedulerClient.java#L106](https://github.com/ow2-proactive/scheduling/blob/3a7a2b2a71a94c2b474b2a32f2ce1cb44b0833bb/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L106): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L125](https://github.com/ow2-proactive/scheduling/blob/3a7a2b2a71a94c2b474b2a32f2ce1cb44b0833bb/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L125): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L146](https://github.com/ow2-proactive/scheduling/blob/3a7a2b2a71a94c2b474b2a32f2ce1cb44b0833bb/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L146): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L166](https://github.com/ow2-proactive/scheduling/blob/3a7a2b2a71a94c2b474b2a32f2ce1cb44b0833bb/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L166): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L186](https://github.com/ow2-proactive/scheduling/blob/3a7a2b2a71a94c2b474b2a32f2ce1cb44b0833bb/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L186): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L209](https://github.com/ow2-proactive/scheduling/blob/3a7a2b2a71a94c2b474b2a32f2ce1cb44b0833bb/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L209): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L230](https://github.com/ow2-proactive/scheduling/blob/3a7a2b2a71a94c2b474b2a32f2ce1cb44b0833bb/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L230): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L247](https://github.com/ow2-proactive/scheduling/blob/3a7a2b2a71a94c2b474b2a32f2ce1cb44b0833bb/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L247): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)
1. ![MINOR][MINOR] [ISchedulerClient.java#L258](https://github.com/ow2-proactive/scheduling/blob/3a7a2b2a71a94c2b474b2a32f2ce1cb44b0833bb/rest/rest-client/src/main/java/org/ow2/proactive/scheduler/rest/ISchedulerClient.java#L258): Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**